packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_row.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+
+import 'logging_controller_v2.dart';
+
+class LoggingTableRow extends StatefulWidget {
+  const LoggingTableRow({
+    super.key,
+    required this.index,
+    required this.data,
+    required this.isSelected,
+  });
+
+  final int index;
+  final LogDataV2 data;
+  final bool isSelected;
+
+  @override
+  State<LoggingTableRow> createState() => _LoggingTableRowState();
+
+  static TextSpan _detailsSpan(String text) {
+    return TextSpan(
+      text: text,
+      style: _detailsStyle,
+    );
+  }
+
+  static TextSpan _metadataSpan(String text) {
+    return TextSpan(
+      text: text,
+      style: _metaDataStyle,
+    );
+  }
```

Comment:
define these directly in the build method below instead of having them be static methods on the widget class. You can just use the TextSpan directly since having methods is just adding an unnecessary level of abstraction:

```
RichText(
  text: TextSpan(widget.data.asLogDetails()),
),

Reply:
These are added so that the same spans can be used in calculating the heights. I wanted to avoid duplicating these parts in case changes get made.